src/dev/flang/parser/Parser.java

```diff
@@ -1101,10 +1101,20 @@ else if (skipDot())
    */
   List<String> argNames()
   {
-    List<String> result = new List<>(name());
+    List<String> result = new List<>();
+    TreeSet<String>  usedNames = new TreeSet<String>();
+    var name = name();
+    result.add(name);
+    usedNames.add(name);
     while (skipComma())
       {
-        result.add(name());
+        var pos = pos();
+        var argName = name();
+        result.add(argName);
+        if (!argName.equals("_") && !usedNames.add(argName))
+          {
+            AstErrors.argumentNamesNotDistinct(posObject(pos));
```

Comment:
argNames is not the best location for this, this will detect `a, a i32` but not `a i32, a i32`. Better check this in `Feature.java`'s constructor that sets `Feature._arguments`.

Reply:
I should have realized this...